---
title: "Summary of ESTIMATE and CIBERSORT"
subtitle: "???<br/>with xaringan"
author: "Grace Yi Chen"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  xaringan::moon_reader:
    chakra: libs/remark-latest.min.js
    css: "my-theme.css"
    lib_dir: libs
    nature:
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
---

```{r setup, include=FALSE}
options(htmltools.dir.version = FALSE)
#xaringan::summon_remark()
```

background-image: url(https://upload.wikimedia.org/wikipedia/commons/b/be/Sharingan_triple.svg)

???

Image credit: [Wikimedia Commons](https://commons.wikimedia.org/wiki/File:Sharingan_triple.svg)

---
class: center, middle

# xaringan

### /???.'ri??.g?n/

---
class: inverse, center, middle

# Get Started

---

## Tumor purity and quantification of tumor-infiltrating immune cells from expression data

- Tumors are complex ecosystems composed of different types of cells. Infiltrating stromal and immune cells form the major fraction of normal cells in tumour tissue and have an important role in cancer biology. Especially, tumor-infiltrating immune cells have a large impact on tumor progression and the success of anti-cancer therapies. 

- Therefore, it is important to study tumor purity and the quantification of tumor-infiltrating immune cells as well as the role of the immune system in human cancers and its involvement in tumor escape mechanisms and response to therapy. 


- Here, I will summarize the overview of two methods: ESTIMATE for tumor purity and CIBERSORT for quantification of tumor-infiltrating immune cells. 

--


---
## General Quantification Method:

- The composition of tumor-infiltrating immune cells can be characterized from bulk tumor RNAseq data using computational approaches based on a set of immune-specific marker genes or expression signatures.

- GSEA approach: The most famous approach for the analysis of maker genes is gene set enrichment analysis (GSEA) . GSEAbased
methods compute an enrichment score (ES) that is high when the genes specific for a certain cell type are amongst the top highly expressed in the sample of interest (i.e., the cell type is enriched in the sample) and low otherwise (Fig.1a)

- Deconvolution approach: Unlike GSEA-based approaches that can only compute a semi-quantitative score, deconvolution methods quantitatively estimate the relative fractions of the cell types. Deconvolution algorithms consider gene expression profiles of a heterogeneous sample as the convolution of the gene expression levels of the different cells, and estimate the unknown cell fractions leveraging on a signature matrix describing the cell-type-specific expression profiles (Fig.1b )


---

![title](.\\summary_slides_files\\figure-html\\Fig1.JPG)

---

.image-70[![title](.\\summary_slides_files\\figure-html\\Table1.JPG)]

---


## ESTIMATE: 
.small[Estimation of STromal and Immune cells in MAlignant Tumours using Expression data:]

- A method that uses gene expression signatures to predict the fraction of stromal and immune cells and tumor purity in tumour samples.

- ESTIMATE algorithm is based on single sample Gene Set Enrichment Analysis (ssGSEA) and generates three scores:

  + stromal score (that captures the presence of stroma in tumor tissue), 
  + immune score (that represents the infiltration of immune cells in tumor tissue), and 
  + estimate score (that infers tumor purity). 
  
- In order to facilitate tumour purity prediction using ESTIMATE signatures, ESTIMATE transformed the scoring system to a [0,1] range. By applying the nonlinear least squares method to the modified TCGA Affymetrix data , ESTIMATE-based tumour purity prediction model was developed.

- The application of ESTIMATE to publicly available microarray expression data sets, as well as new microarray or RNA-seq-based transcriptome profiles, may help in elucidating the facilitating roles of the microenvironment to neoplastic cell and provide new insights into context in which genomic alterations occur.

---
### An overview of the ESTIMATE algorithm

![title](.\\summary_slides_files\\figure-html\\Fig2.JPG)

---

## Limitation:

- Because of the dependency of ESTIMATE on infiltrating stromal and immune cells, one of the limitation is its inability to accurately infer tumour cellularity of hematopoietic or stromal tumours (for example, leukaemia and gastrointestinal stromal tumours) because of the high and tumour-intrinsic expression of stromalor immune-related genes. 

- Owing to the lack of data, we were unable to evaluate ESTIMATE in the context of tumour types such as prostate or pancreas cancer that may present with atypical patterns of tumour-associated cells?that is, increased fractions of normal epithelial cells.

---


### ESTIMATE: Application

- R package: ESTIMATE; predicts the presence of stromal and immune cells in tumor tissue using gene expression
data.
```{r eval=FALSE, tidy=FALSE}
library(utils)
rforge <- "http://r-forge.r-project.org"
install.packages("estimate", repos=rforge, dependencies=TRUE)
```

#### Steps:

1: Gene expression data
    * 10 ovarian cancer samples using Affymetrix U133Plus2.0 platform. It has gene-level expression data consisting 17,256 genes (row) and 10 samples (column).

2. Second, we unify different number of genes per microarray platforms to 10,412 common genes. This algorithm uses HUGO GeneSymbols or Entrez gene IDs. Therefore, the row name in an input data must be gene symbols or Entrez gene IDs.

3. We calculate stromal and immune scores that represent the presence of stromal and immune cells in tumor tissue, respectively. In addition, this function measures raw ESTIMATE score that are positively correlated with DNA copy number based tumor purity.

---

## CIBERSORT

- A deconvolution method for characterizing cell composition of complex tissues from their gene expression profiles;

- Requires an input matrix of reference gene expression signatures, collectively used to estimate the relative proportions of each cell type of interest. 

- To deconvolve the mixture, linear support vector regression (SVR) is applied which is highly robust with respect to noise. SVR performs a feature selection, in which genes from the signature matrix are adaptively selected to deconvolve a given mixture. An empirically defined global P value for the deconvolution is then determined

- When applied with statistical filtration, CIBERSORT, designed using LM22 (it contains 547 genes that distinguish 22 human hematopoietic cell phenotypes, including seven T-cell types, na?ve and memory B cells, plasma cells, natural killer (NK) cells and myeloid subsets) allows for highly sensitive and specific discrimination of human leukocyte subsets. 

- We note that our filtration approach is likely applicable to other signature matrices and other GEP deconvolution methods.
---

### An overview of the CIBERSORT algorithm

![title](.\\summary_slides_files\\figure-html\\Fig3.JPG)

---

### CIBERSORT: Limitation

- The most significant current limitation of CIBERSORT, and indeed all signature gene?based methods, is the fidelity of reference profiles, which could deviate in cells undergoing heterotypic interactions, phenotypic plasticity or disease-induced dysregulation.

- Second, CIBERSORT currently does not provide P values for detection limits of individual cell types.

- Third, despite CIBERSORT showing a considerably lower estimation bias than other approaches, it systematically over- or underestimated some cell types;

- Finally, although CIBERSORT was not explicitly tested on RNA-seq data, the linearity assumptions made by our method are likely to hold, as previously suggested. (could be normalized with limma voom)

- Validated on microarray data of cell mixtures derived from blood and from lymph node biopsies, CIBERSORT proved to have a high accuracy in the simultaneous deconvolution of nine and three immune cell subsets, respectively, whereas it showed a lower accuracy in the quantification of gamma-delta T cells

---


### CIBERSORT: Application

#### Example and [tutorial](https://cibersort.stanford.edu/tutorial.php) 
Use LM22 as signature gene matrix as example.

1. Prepare mixture file: tab-deliminated, HUGO symbols in column 1; Mixture labesl in row 1; in non-log space; will by default normalized by quantile normalization

![title](.\\summary_slides_files\\figure-html\\Fig4.JPG)


2. upload data and run the job
---
#### Example result:
.image-50[![title](.\\summary_slides_files\\figure-html\\Fig5.JPG)]
.image-50[![title](.\\summary_slides_files\\figure-html\\Fig6.JPG)]

---

#### Custom Signature Genes File:

1. Prepare your reference sample file:
.image-50[![title](.\\summary_slides_files\\figure-html\\Fig7.JPG)]

2. Prepare penotype classes file:
.image-50[![title](.\\summary_slides_files\\figure-html\\Fig8.JPG)]

3. upload data and run the job


---
